packages/api/src/utils/url.js

```diff
@@ -25,9 +29,12 @@ export function getSourceUrl(request, env) {
       `invalid URL provided: ${request.params.url}: maximum allowed length or URL is ${MAX_ALLOWED_URL_LENGTH}`
     )
   }
-  if (!urlString.includes(env.GATEWAY_DOMAIN)) {
+  if (
+    !env.GATEWAY_DOMAINS.filter((gwDomain) => urlString.includes(gwDomain))
+      .length
```

Comment:
```suggestion
    !env.GATEWAY_DOMAINS.includes(new URL(urlString).hostname)
```

Reply:
```js
const a = new URL('http://bafkreidyeivj7adnnac6ljvzj2e3rd5xdw3revw4da7mx2ckrstapoupoq.ipfs.localhost:9081')
a.hostname
'bafkreidyeivj7adnnac6ljvzj2e3rd5xdw3revw4da7mx2ckrstapoupoq.ipfs.localhost'
```

URLs are tricky, we can't go the way suggested here

Reply:
Just `url.host` and `.endsWith` instead? I think it's better to use the domain if we can!